# Stack Overflow 2022 Survey Analysis
This is an analysis of the 2022 Stack Overflow Survey. The survey has more than 70,000 respondents but I only chose to analyze the submissions of respondents from the Philippines.

In [4]:
import pandas as pd
dataset = pd.read_csv("data/survey_results_public.csv")
dataset

ResponseId                                         MainBranch  \
0               1                                      None of these   
1               2                     I am a developer by profession   
2               3  I am not primarily a developer, but I write co...   
3               4                     I am a developer by profession   
4               5                     I am a developer by profession   
...           ...                                                ...   
73263       73264                     I am a developer by profession   
73264       73265                     I am a developer by profession   
73265       73266  I am not primarily a developer, but I write co...   
73266       73267                     I am a developer by profession   
73267       73268  I used to be a developer by profession, but no...   

                                              Employment  \
0                                                    NaN   
1                                    Employed, full-time   
2                                    Employed, full-time   
3                                    Employed, full-time   
4                                    Employed, full-time   
...                                                  ...   
73263                                Employed, full-time   
73264                                Employed, full-time   
73265                                Employed, full-time   
73266                                Employed, full-time   
73267  Independent contractor, freelancer, or self-em...   

                                 RemoteWork  \
0                                       NaN   
1                              Fully remote   
2      Hybrid (some remote, some in-person)   
3                              Fully remote   
4      Hybrid (some remote, some in-person)   
...                                     ...   
73263                          Fully remote   
73264                        Full in-person   
73265  Hybrid (some remote, some in-person)   
73266  Hybrid (some remote, some in-person)   
73267                          Fully remote   

                                        CodingActivities  \
0                                                    NaN   
1               Hobby;Contribute to open-source projects   
2                                                  Hobby   
3                           I don’t code outside of work   
4                                                  Hobby   
...                                                  ...   
73263                            Freelance/contract work   
73264                                              Hobby   
73265                      Hobby;School or academic work   
73266                                              Hobby   
73267  Hobby;Contribute to open-source projects;Boots...   

                                               EdLevel  \
0                                                  NaN   
1                                                  NaN   
2      Master’s degree (M.A., M.S., M.Eng., MBA, etc.)   
3         Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
4         Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
...                                                ...   
73263     Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
73264  Master’s degree (M.A., M.S., M.Eng., MBA, etc.)   
73265     Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
73266     Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
73267     Bachelor’s degree (B.A., B.S., B.Eng., etc.)   

                                               LearnCode  \
0                                                    NaN   
1                                                    NaN   
2      Books / Physical media;Friend or family member...   
3      Books / Physical media;School (i.e., Universit...   
4      Other online resources (e.g., videos, blogs, f...   
...                                                  ...   
73263  Books / Phy

In [2]:
schema = pd.read_csv("data/survey_results_schema.csv")
schema

qid        qname                                           question  \
0    QID16           S0  <div><span style="font-size:19px;"><strong>Hel...   
1    QID12     MetaInfo                                  Browser Meta Info   
2     QID1           S1  <span style="font-size:22px; font-family: aria...   
3     QID2   MainBranch  Which of the following options best describes ...   
4   QID296   Employment  Which of the following best describes your cur...   
..     ...          ...                                                ...   
74  QID290  Frequency_2  Interacting with people outside of your immedi...   
75  QID290  Frequency_3  Encountering knowledge silos (where one indivi...   
76  QID294  TrueFalse_1  Are you involved in supporting new hires durin...   
77  QID294  TrueFalse_2  Do you use learning resources provided by your...   
78  QID294  TrueFalse_3  Does your employer give you time to learn new ...   

   force_resp  type selector  
0       False    DB       TB  
1       False  Meta  Browser  
2       False    DB       TB  
3        True    MC     SAVR  
4       False    MC     MAVR  
..        ...   ...      ...  
74        NaN    MC     MAVR  
75        NaN    MC     MAVR  
76        NaN    MC     MAVR  
77        NaN    MC     MAVR  
78        NaN    MC     MAVR  

[79 rows x 6 columns]

In [3]:
list(dataset.columns)

['ResponseId',
 'MainBranch',
 'Employment',
 'RemoteWork',
 'CodingActivities',
 'EdLevel',
 'LearnCode',
 'LearnCodeOnline',
 'LearnCodeCoursesCert',
 'YearsCode',
 'YearsCodePro',
 'DevType',
 'OrgSize',
 'PurchaseInfluence',
 'BuyNewTool',
 'Country',
 'Currency',
 'CompTotal',
 'CompFreq',
 'LanguageHaveWorkedWith',
 'LanguageWantToWorkWith',
 'DatabaseHaveWorkedWith',
 'DatabaseWantToWorkWith',
 'PlatformHaveWorkedWith',
 'PlatformWantToWorkWith',
 'WebframeHaveWorkedWith',
 'WebframeWantToWorkWith',
 'MiscTechHaveWorkedWith',
 'MiscTechWantToWorkWith',
 'ToolsTechHaveWorkedWith',
 'ToolsTechWantToWorkWith',
 'NEWCollabToolsHaveWorkedWith',
 'NEWCollabToolsWantToWorkWith',
 'OpSysProfessional use',
 'OpSysPersonal use',
 'VersionControlSystem',
 'VCInteraction',
 'VCHostingPersonal use',
 'VCHostingProfessional use',
 'OfficeStackAsyncHaveWorkedWith',
 'OfficeStackAsyncWantToWorkWith',
 'OfficeStackSyncHaveWorkedWith',
 'OfficeStackSyncWantToWorkWith',
 'Blockchain',
 'NEWSOSites

In [5]:
survey = dataset.copy()
survey["Country"].value_counts()

United States of America                                13543
India                                                    6639
Germany                                                  5395
United Kingdom of Great Britain and Northern Ireland     4190
Canada                                                   2490
                                                        ...  
Monaco                                                      1
Djibouti                                                    1
Seychelles                                                  1
Solomon Islands                                             1
Saint Kitts and Nevis                                       1
Name: Country, Length: 180, dtype: int64

## Filtering the dataset
Getting only the submissions of respondents from the Philippines

In [6]:
survey_ph = survey[survey["Country"] == "Philippines"]
pd.set_option('display.max_columns', None)
survey_ph

ResponseId                                         MainBranch  \
177           178  I used to be a developer by profession, but no...   
1586         1587                     I am a developer by profession   
2222         2223                     I am a developer by profession   
2563         2564                     I am a developer by profession   
3039         3040                              I am learning to code   
...           ...                                                ...   
72615       72616                     I am a developer by profession   
72890       72891                     I am a developer by profession   
72967       72968                     I am a developer by profession   
73227       73228                     I am a developer by profession   
73252       73253                              I am learning to code   

                                              Employment  \
177                                  Employed, full-time   
1586                                 Employed, full-time   
2222                                 Employed, full-time   
2563   Employed, full-time;Independent contractor, fr...   
3039                  Not employed, but looking for work   
...                                                  ...   
72615             Employed, full-time;Student, part-time   
72890                                Employed, full-time   
72967  Independent contractor, freelancer, or self-em...   
73227                                Employed, full-time   
73252                                 Student, part-time   

                                 RemoteWork               CodingActivities  \
177                            Fully remote   I don’t code outside of work   
1586                           Fully remote                          Hobby   
2222                           Fully remote                          Hobby   
2563                           Fully remote                          Hobby   
3039                                    NaN                            NaN   
...                                     ...                            ...   
72615                          Fully remote  Hobby;School or academic work   
72890  Hybrid (some remote, some in-person)                          Hobby   
72967  Hybrid (some remote, some in-person)  Hobby;Freelance/contract work   
73227                          Fully remote                          Hobby   
73252                                   NaN                            NaN   

                                                 EdLevel  \
177         Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
1586   Some college/university study without earning ...   
2222   Some college/university study without earning ...   
2563        Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
3039        Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
...                                                  ...   
72615    Master’s degree (M.A., M.S., M.Eng., MBA, etc.)   
72890       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
72967       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
73227                Associate degree (A.A., A.S., etc.)   
73252       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   

                                               LearnCode  \
177    Other online resources (e.g., videos, blogs, f...   
1586   Books / Physical media;Other online resources ...   
2222   Other online resources (e.g., videos, blogs, f...   
2563   Other online resources (e.g., videos, blogs, f...   
3039   Other online resources (e.g., videos, blogs, f...   
...                                                  ...   
72615  Books / Physical media;Other online resources ...   
72890  Other online resources (e.g., videos, blogs, f...   
72967  Other online resources (e.g., videos, blogs, f...   
73227  Other online resources (e.g., videos, blogs, f...   
73252  Other online resources (e.g., videos, blogs, f...   

                                         L

There are 366 respondents from the Philippines in the Stack Overflow 2022 Survey

## Basic Information


In [6]:
survey_ph["MainBranch"].value_counts()

I am a developer by profession                                                   222
I am learning to code                                                             72
I code primarily as a hobby                                                       40
I am not primarily a developer, but I write code sometimes as part of my work     24
I used to be a developer by profession, but no longer am                           8
Name: MainBranch, dtype: int64

In [7]:
survey_ph["MainBranch"].value_counts(normalize=True)

I am a developer by profession                                                   0.606557
I am learning to code                                                            0.196721
I code primarily as a hobby                                                      0.109290
I am not primarily a developer, but I write code sometimes as part of my work    0.065574
I used to be a developer by profession, but no longer am                         0.021858
Name: MainBranch, dtype: float64

222 people or ~60% of respondents from the Philippines are developers by profession.

In [17]:
survey_ph["Age"].value_counts()

25-34 years old       167
18-24 years old       118
Under 18 years old     36
35-44 years old        30
45-54 years old         8
Prefer not to say       3
55-64 years old         1
65 years or older       1
Name: Age, dtype: int64

In [7]:
survey_ph["Age"].value_counts(normalize=True)

25-34 years old       0.458791
18-24 years old       0.324176
Under 18 years old    0.098901
35-44 years old       0.082418
45-54 years old       0.021978
Prefer not to say     0.008242
55-64 years old       0.002747
65 years or older     0.002747
Name: Age, dtype: float64

People aged 25 - 34 years old represents almost 46% of the respondents, and people aged 18 - 24 years old represents 32%. 10% of the respondents are under 18 years of age while the remaining are aged 35 and above.

In [18]:
survey_ph["Gender"].value_counts()

Man                                                        320
Woman                                                       26
Prefer not to say                                            7
Non-binary, genderqueer, or gender non-conforming            4
Or, in your own words:                                       2
Man;Or, in your own words:                                   2
Man;Non-binary, genderqueer, or gender non-conforming        1
Woman;Non-binary, genderqueer, or gender non-conforming      1
Name: Gender, dtype: int64

In [8]:
survey_ph["Gender"].value_counts(normalize=True)

Man                                                        0.881543
Woman                                                      0.071625
Prefer not to say                                          0.019284
Non-binary, genderqueer, or gender non-conforming          0.011019
Or, in your own words:                                     0.005510
Man;Or, in your own words:                                 0.005510
Man;Non-binary, genderqueer, or gender non-conforming      0.002755
Woman;Non-binary, genderqueer, or gender non-conforming    0.002755
Name: Gender, dtype: float64

Majority of the respondents from the Philippines are men. 88% are men, 7% are women, and the remaining are part of the LGBT community.

In [10]:
sph_employment = survey_ph["Employment"].str.split(";", expand=True)
sph_employment.stack().value_counts()

Employed, full-time                                     202
Student, full-time                                       69
Independent contractor, freelancer, or self-employed     62
Not employed, but looking for work                       24
Employed, part-time                                      24
Student, part-time                                       19
Not employed, and not looking for work                   19
I prefer not to say                                      14
Retired                                                   1
dtype: int64

## Education, work, and career

In [10]:
survey_ph["EdLevel"].value_counts()

Bachelor’s degree (B.A., B.S., B.Eng., etc.)                                          228
Secondary school (e.g. American high school, German Realschule or Gymnasium, etc.)     52
Some college/university study without earning a degree                                 42
Primary/elementary school                                                              18
Master’s degree (M.A., M.S., M.Eng., MBA, etc.)                                        13
Something else                                                                          7
Associate degree (A.A., A.S., etc.)                                                     5
Professional degree (JD, MD, etc.)                                                      1
Name: EdLevel, dtype: int64

In [11]:
survey_ph["EdLevel"].value_counts(normalize=True)

Bachelor’s degree (B.A., B.S., B.Eng., etc.)                                          0.622951
Secondary school (e.g. American high school, German Realschule or Gymnasium, etc.)    0.142077
Some college/university study without earning a degree                                0.114754
Primary/elementary school                                                             0.049180
Master’s degree (M.A., M.S., M.Eng., MBA, etc.)                                       0.035519
Something else                                                                        0.019126
Associate degree (A.A., A.S., etc.)                                                   0.013661
Professional degree (JD, MD, etc.)                                                    0.002732
Name: EdLevel, dtype: float64

62% has a bachelor's degree.

In [12]:
survey_ph["LearnCode"].value_counts()

Other online resources (e.g., videos, blogs, forum)                                                                                                       31
Other online resources (e.g., videos, blogs, forum);School (i.e., University, College, etc)                                                               22
School (i.e., University, College, etc)                                                                                                                   18
Other online resources (e.g., videos, blogs, forum);School (i.e., University, College, etc);Online Courses or Certification                               17
Books / Physical media;Other online resources (e.g., videos, blogs, forum);School (i.e., University, College, etc);Online Courses or Certification        16
                                                                                                                                                          ..
Books / Physical media;Other online resources (e.g., video

In [16]:
sph_learnCode = survey_ph["LearnCode"].str.split(";", expand=True)
sph_learnCode.stack().value_counts()

Other online resources (e.g., videos, blogs, forum)    288
School (i.e., University, College, etc)                265
Online Courses or Certification                        194
Books / Physical media                                 149
On the job training                                    103
Coding Bootcamp                                         76
Colleague                                               70
Friend or family member                                 40
Hackathons (virtual or in-person)                       39
Other (please specify):                                 22
dtype: int64

Most common ways of learning how to code are through online resources and through school.

In [41]:
sph_devType = survey_ph["DevType"].str.split(";", expand=True)
sph_devType.stack().value_counts()

Developer, full-stack                            150
Developer, back-end                              108
Developer, front-end                              97
Developer, desktop or enterprise applications     44
Developer, mobile                                 44
Database administrator                            33
System administrator                              26
Student                                           24
DevOps specialist                                 23
Data or business analyst                          20
Cloud infrastructure engineer                     18
Developer, embedded applications or devices       18
Designer                                          17
Educator                                          16
Other (please specify):                           16
Project manager                                   16
Engineer, data                                    13
Developer, QA or test                             13
Engineering manager                           

Full-stack, back-end, and front-end developers are the most common developer types.

In [14]:
survey_ph["OrgSize"].value_counts()

100 to 499 employees                                  51
20 to 99 employees                                    45
10 to 19 employees                                    23
10,000 or more employees                              23
2 to 9 employees                                      20
1,000 to 4,999 employees                              18
Just me - I am a freelancer, sole proprietor, etc.    14
500 to 999 employees                                  12
I don’t know                                           9
5,000 to 9,999 employees                               4
Name: OrgSize, dtype: int64

In [15]:
survey_ph["OrgSize"].value_counts(normalize=True)

100 to 499 employees                                  0.232877
20 to 99 employees                                    0.205479
10 to 19 employees                                    0.105023
10,000 or more employees                              0.105023
2 to 9 employees                                      0.091324
1,000 to 4,999 employees                              0.082192
Just me - I am a freelancer, sole proprietor, etc.    0.063927
500 to 999 employees                                  0.054795
I don’t know                                          0.041096
5,000 to 9,999 employees                              0.018265
Name: OrgSize, dtype: float64

In [12]:
survey_ph["CompFreq"].value_counts()

Monthly    159
Yearly      18
Weekly       5
Name: CompFreq, dtype: int64

In [13]:
sph_salaryIsYearly = survey_ph[survey_ph["CompFreq"] == "Yearly"]
sph_salaryIsYearly

ResponseId                                         MainBranch  \
2563         2564                     I am a developer by profession   
4064         4065                     I am a developer by profession   
5463         5464  I am not primarily a developer, but I write co...   
5885         5886                     I am a developer by profession   
8061         8062                     I am a developer by profession   
21987       21988                     I am a developer by profession   
22391       22392                     I am a developer by profession   
22615       22616                     I am a developer by profession   
30327       30328                     I am a developer by profession   
31762       31763                     I am a developer by profession   
32028       32029                     I am a developer by profession   
32206       32207                     I am a developer by profession   
53574       53575                     I am a developer by profession   
54767       54768  I am not primarily a developer, but I write co...   
62160       62161                     I am a developer by profession   
65142       65143  I am not primarily a developer, but I write co...   
67155       67156                     I am a developer by profession   
67159       67160                     I am a developer by profession   

                                              Employment  \
2563   Employed, full-time;Independent contractor, fr...   
4064                                 Employed, full-time   
5463                                 Employed, full-time   
5885                                 Employed, full-time   
8061                                 Employed, full-time   
21987                                Employed, full-time   
22391  Independent contractor, freelancer, or self-em...   
22615            Employed, full-time;Employed, part-time   
30327                                Employed, full-time   
31762  Independent contractor, freelancer, or self-em...   
32028                                Employed, full-time   
32206                                Employed, full-time   
53574                                Employed, full-time   
54767                                Employed, full-time   
62160                                Employed, full-time   
65142  Employed, full-time;Independent contractor, fr...   
67155                                Employed, full-time   
67159                                Employed, full-time   

                                 RemoteWork  \
2563                           Fully remote   
4064   Hybrid (some remote, some in-person)   
5463                           Fully remote   
5885                           Fully remote   
8061                           Fully remote   
21987                          Fully remote   
22391                          Fully remote   
22615                        Full in-person   
30327                          Fully remote   
31762                          Fully remote   
32028                          Fully remote   
32206                          Fully remote   
53574  Hybrid (some remote, some in-person)   
54767  Hybrid (some remote, some in-person)   
62160                          Fully remote   
65142                          Fully remote   
67155                          Fully remote   
67159                          Fully remote   

                                        CodingActivities  \
2563                                               Hobby   
4064                                               Hobby   
5463   Hobby;Contribute to open-source projects;Freel...   
5885                        I don’t code outside of work   
8061                       Hobby;Freelance/contract work   
21987  Hobby;Contribute to open-source projects;Boots...   
22391                           Bootstrapping a business   
22615                            Freelance/contract work   
30327  Hobby;Bootstrapping a business;Freelance/contr...   
317

In [16]:
sph_salaryIsYearly["CompTotal"].max() / 12

416666.6666666667

In [22]:
sph_salaryIsMonthly = survey_ph[survey_ph["CompFreq"] == "Monthly"]
sph_salaryIsMonthly

ResponseId                      MainBranch  \
1586         1587  I am a developer by profession   
2222         2223  I am a developer by profession   
3273         3274  I am a developer by profession   
3611         3612  I am a developer by profession   
4274         4275  I am a developer by profession   
...           ...                             ...   
72385       72386  I am a developer by profession   
72466       72467  I am a developer by profession   
72547       72548  I am a developer by profession   
72890       72891  I am a developer by profession   
73227       73228  I am a developer by profession   

                                              Employment  \
1586                                 Employed, full-time   
2222                                 Employed, full-time   
3273                                 Employed, full-time   
3611   Independent contractor, freelancer, or self-em...   
4274                                 Employed, full-time   
...                                                  ...   
72385                                Employed, full-time   
72466  Employed, full-time;Independent contractor, fr...   
72547                                Employed, full-time   
72890                                Employed, full-time   
73227                                Employed, full-time   

                                 RemoteWork  \
1586                           Fully remote   
2222                           Fully remote   
3273   Hybrid (some remote, some in-person)   
3611                           Fully remote   
4274                           Fully remote   
...                                     ...   
72385                          Fully remote   
72466                          Fully remote   
72547                          Fully remote   
72890  Hybrid (some remote, some in-person)   
73227                          Fully remote   

                                        CodingActivities  \
1586                                               Hobby   
2222                                               Hobby   
3273            Hobby;Contribute to open-source projects   
3611   Hobby;Bootstrapping a business;Freelance/contr...   
4274                                               Hobby   
...                                                  ...   
72385                      Hobby;Other (please specify):   
72466    Freelance/contract work;Other (please specify):   
72547                                              Hobby   
72890                                              Hobby   
73227                                              Hobby   

                                                 EdLevel  \
1586   Some college/university study without earning ...   
2222   Some college/university study without earning ...   
3273        Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
3611        Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
4274        Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
...                                                  ...   
72385       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
72466       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
72547  Secondary school (e.g. American high school, G...   
72890       Bachelor’s degree (B.A., B.S., B.Eng., etc.)   
73227                Associate degree (A.A., A.S., etc.)   

                                               LearnCode  \
1586   Books / Physical media;Other online resources ...   
2222   Other online resources (e.g., videos, blogs, f...   
3273   Books / Physical media;Other online resources ...   
3611   Books / Physical media;Other online resources ...   
4274   Friend or family member;Other online resources...   
...                                                  ...   
72385  Books / Physical media;Friend or family member...   
72466  Other online resources (e.g., videos, blogs, f...   
72547  Other online resources (e.g., videos, blogs, f...   
72890  Other online resources (e.g., video

In [23]:
sph_salaryIsMonthly["CompTotal"].max()

1400000.0

The highest monthly salary is P1,400,000.

In [32]:
survey_ph.loc[sph_salaryIsMonthly["CompTotal"].idxmax()]

ResponseId                                                                    29225
MainBranch                                           I am a developer by profession
Employment                                                      Employed, full-time
RemoteWork                                                             Fully remote
CodingActivities                                       I don’t code outside of work
EdLevel                                Bachelor’s degree (B.A., B.S., B.Eng., etc.)
LearnCode                                   School (i.e., University, College, etc)
LearnCodeOnline                                                                 NaN
LearnCodeCoursesCert                                                            NaN
YearsCode                                                                        14
YearsCodePro                                                                     10
DevType                                                         Developer, b

The person who has the highest monthly salary among the respondents is a fulltime employee and works in a fully remote setting. He ha a bachelor's degree and is working as a backend developer. He has been coding professionally for a decade and he's aged 25-34 years old. The languages he's worked with are HTML/CSS, JavaScript, Python, and SQL.

In [38]:
sph_languages = survey_ph["LanguageHaveWorkedWith"].str.split(";", expand=True)
sph_languages.stack().value_counts()

JavaScript     266
HTML/CSS       241
SQL            190
Python         157
TypeScript     133
PHP            129
Java           118
C#              96
Bash/Shell      70
C++             64
C               52
PowerShell      33
Kotlin          32
Dart            28
VBA             26
Go              22
Assembly        22
Swift           20
Rust            16
Ruby            13
MATLAB          13
Objective-C      8
Lua              7
Groovy           7
Delphi           7
Perl             6
Elixir           4
Solidity         4
Scala            4
Haskell          4
R                3
Clojure          3
APL              2
Crystal          2
LISP             2
COBOL            2
SAS              2
Julia            1
dtype: int64

In [42]:
sph_database = survey_ph["DatabaseHaveWorkedWith"].str.split(";", expand=True)
sph_database.stack().value_counts()

MySQL                         191
PostgreSQL                    121
SQLite                        113
MongoDB                        93
Microsoft SQL Server           84
MariaDB                        76
Redis                          57
Firebase Realtime Database     35
Oracle                         33
Cloud Firestore                32
DynamoDB                       18
Elasticsearch                  13
IBM DB2                         6
Couchbase                       5
Cassandra                       5
CouchDB                         3
Neo4j                           3
dtype: int64

In [43]:
sph_cloudPlatform = survey_ph["PlatformHaveWorkedWith"].str.split(";", expand=True)
sph_cloudPlatform.stack().value_counts()

AWS                            115
Firebase                        81
Heroku                          79
Google Cloud                    74
Microsoft Azure                 49
DigitalOcean                    47
VMware                          21
Linode                          13
Managed Hosting                 13
Oracle Cloud Infrastructure      5
OVH                              4
OpenStack                        1
dtype: int64

In [45]:
sph_webFramework = survey_ph["WebframeHaveWorkedWith"].str.split(";", expand=True)
sph_webFramework.stack().value_counts()

Node.js           156
React.js          133
jQuery            107
Express            85
Vue.js             73
Angular            66
Laravel            63
ASP.NET            55
Next.js            49
ASP.NET Core       46
Angular.js         43
Django             43
Flask              38
Nuxt.js            22
Svelte             14
Ruby on Rails      13
Symfony            11
Blazor             11
FastAPI            10
Fastify             7
Phoenix             7
Drupal              5
Deno                3
Gatsby              3
Play Framework      2
dtype: int64

In [46]:
sph_devTools = survey_ph["ToolsTechHaveWorkedWith"].str.split(";", expand=True)
sph_devTools.stack().value_counts()

npm              205
Docker           129
Yarn              86
Homebrew          51
Unity 3D          38
Kubernetes        35
Ansible           17
Terraform         17
Unreal Engine     12
Puppet             6
Chef               5
Flow               3
Pulumi             2
dtype: int64

In [48]:
sph_devEnvironment = survey_ph["NEWCollabToolsHaveWorkedWith"].str.split(";", expand=True)
sph_devEnvironment.stack().value_counts()

Visual Studio Code                  278
Visual Studio                       137
Sublime Text                         99
Notepad++                            99
Android Studio                       98
IntelliJ                             81
Vim                                  72
PyCharm                              64
Eclipse                              50
NetBeans                             47
Atom                                 45
Xcode                                41
Nano                                 30
Neovim                               29
IPython/Jupyter                      29
PhpStorm                             28
Webstorm                             17
Rider                                11
Emacs                                 9
Qt Creator                            8
CLion                                 8
Spyder                                6
RAD Studio (Delphi, C++ Builder)      6
GoLand                                2
TextMate                              2


In [51]:
sph_OSPro = survey_ph["OpSysProfessional use"].str.split(";", expand=True)
sph_OSPro.stack().value_counts()

Windows                              188
Linux-based                          129
macOS                                110
Windows Subsystem for Linux (WSL)     62
BSD                                   10
Other (please specify):                5
dtype: int64

In [52]:
sph_OSPersonal = survey_ph["OpSysPersonal use"].str.split(";", expand=True)
sph_OSPersonal.stack().value_counts()

Windows                              282
Linux-based                          126
macOS                                 76
Windows Subsystem for Linux (WSL)     69
BSD                                    7
Other (please specify):                5
dtype: int64

In [53]:
sph_vc = survey_ph["VersionControlSystem"].str.split(";", expand=True)
sph_vc.stack().value_counts()

Git                        322
I don't use one             32
SVN                         20
Other (please specify):     11
dtype: int64

## Stack Community

In [54]:
survey_ph["SOVisitFreq"].value_counts()

Daily or almost daily                  109
Multiple times per day                  86
A few times per week                    84
A few times per month or weekly         64
Less than once per month or monthly     16
Name: SOVisitFreq, dtype: int64

In [55]:
survey_ph["TimeSearching"].value_counts()

30-60 minutes a day           69
15-30 minutes a day           49
60-120 minutes a day          19
Over 120 minutes a day        14
Less than 15 minutes a day    11
Name: TimeSearching, dtype: int64